### ISSUE TITLE
Custom CSS not overriding plus conflict issues

### ISSUE THREAD

**Issue by longhornrumble** [aws-samples/aws-lex-web-ui#702]
Hi,

I have my custom CSS file loaded as the last link in the head as instructed (see attached index file). Unfortunately, the network view (attached file) in Dev Tools says that it IS loading as the last head item, but there are other CSS files coming after that are being loaded from the loader.js file that's scripted in the index (also in attached index file).

No matter what I do, the custom CSS is being overridden by other CSS files (I've market !Important in key areas as well). Isn't the custom file supposed to override them or is it necessary to go through the various CSS files and make changes to them directly?

Finally, there's a screenshot that shows some warnings from the console in Dev Tools. I'm guessing the most important one is the "global variable loader seems to already exist" warning. I guess that the parent and index files are calling the same scripts (loader primarily) and that these are in conflict with one another.

I'm fresh out of ideas on how to remedy this and could really use some help. Thank you! 

The bot works, but the look and feel need to be corrected. 

![Index](https://github.com/aws-samples/aws-lex-web-ui/assets/64515532/9e8f5675-0753-4109-810d-9bd4ab7dc534)
![Network](https://github.com/aws-samples/aws-lex-web-ui/assets/64515532/75a14ec1-fa32-43ae-ad34-b8a210c1944b)
![Conflicts](https://github.com/aws-samples/aws-lex-web-ui/assets/64515532/59660cc9-e334-4e0b-8fbd-2e670756f850)

**Comment by atjohns** [comment]
Hi,

Yes if you mark your elements as !important they should override any existing styles. Do you mind sharing your custom-chatbot-styles.css file contents?

**Comment by longhornrumble** [comment]
Not at all. Thank you. 

I need to clean up the CSS, but you can see the gist. 

As an area of focus, look at the title of the bot. I'm trying to decrease the font size to get it to fit. I've attached a look at Styles from Dev Tools where you can see this area (H2). The CSS isn't affecting the current settings. There are many other areas that are being governed by some other css file that are overriding the custom file. 

[customcss.txt](https://github.com/user-attachments/files/16090692/customcss.txt)
![Bot Title](https://github.com/aws-samples/aws-lex-web-ui/assets/64515532/2fae6b8c-00ab-4a53-861f-aa60f4b4072e)
![Styles](https://github.com/aws-samples/aws-lex-web-ui/assets/64515532/afd1fd92-c91c-45a6-968e-b69b2d250c7a)

**Comment by atjohns** [comment]
Thanks I will take a look and see what is happening here

**Comment by longhornrumble** [comment]
Thank you! 

I cleaned up the CSS a bit. New txt doc attached. 

Also, the file that is failing to be overridden for the Title of the bot is the 3.4.6 Vuetify.min.css file (screenshot). Because all of these CSS files are loading after the head CSS files in the index, they are applying their default values. 

I have an iFrame loading so the loader.js file is loading per the way it's coded in the index.html file (in the /body). 

Appreciate any suggestions you can muster.

![Screenshot 2024-07-03 at 5 57 28 PM](https://github.com/aws-samples/aws-lex-web-ui/assets/64515532/1252c958-9dc7-4256-aa6a-8ac19d245024)
[customcss.txt](https://github.com/user-attachments/files/16091474/customcss.txt)

**Comment by israelias** [comment]
@longhornrumble In case it's quick can you enable the menu icon just for the sake of isolating other variables that pop in to that toolbar between icon and title that could boggle everything (should you opt for it later or otherwise)

**Comment by longhornrumble** [comment]
The minimize icon? If there’s a menu icon, I didn’t know it existed. Is it
in the config file?

**Comment by israelias** [comment]
@longhornrumble Yeah it's the Toolbar menu. You can enable through `x` path depending on how you launched.

But while we're up there, I'd test three features that can populate that row as distinct html elements that behave differently and/or ship styled and positioned where they are without apology (barring`!important` overrides placed where they must, of course)


**1. Toolbar Menu** [looks like](https://github.com/aws-samples/aws-lex-web-ui/blob/master/img/feedbackButtons.png)

Toggling any of the following three `ui.saveHistory`, `ui.enableLiveChat`, `ui.enableLogin` will give you a menu with a dropdown to execute a number of what those offer in case they are plugged to play in your backend. Except SaveHistory, which is the easy one to test for the css situation (it just persists the chat window messages by saving and subscribing to sessionStorage `sessionStorage.setItem('store', JSON.stringify(state));` so I would try that to see the css consequences of Toolbar Menu.

- https://github.com/aws-samples/aws-lex-web-ui/blob/f44b371b370bd40334afa6cf11222d0677e3ad1f/src/config/default-lex-web-ui-loader-config.json#L52
- https://github.com/aws-samples/aws-lex-web-ui/blob/f44b371b370bd40334afa6cf11222d0677e3ad1f/src/config/default-lex-web-ui-loader-config.json#L53
- https://github.com/aws-samples/aws-lex-web-ui/blob/f44b371b370bd40334afa6cf11222d0677e3ad1f/src/config/default-lex-web-ui-loader-config.json#L39


**2. Help Intent** [quick ref](https://github.com/aws-samples/aws-lex-web-ui#new-features-in-version-0150)

- https://github.com/aws-samples/aws-lex-web-ui/blob/f44b371b370bd40334afa6cf11222d0677e3ad1f/src/config/default-lex-web-ui-loader-config.json#L37


**3. Back Button** [image ref](https://github.com/aws-samples/aws-lex-web-ui/blob/master/img/toolbar.png)

This guy animates and will correctly disappear if there's nothing to go back to. 

- https://github.com/aws-samples/aws-lex-web-ui/blob/f44b371b370bd40334afa6cf11222d0677e3ad1f/src/config/default-lex-web-ui-loader-config.json#L49

___

[**extra** (for image only)](https://catalog.us-east-1.prod.workshops.aws/workshops/20c56f9e-9c0a-4174-a661-9f40d9f063ac/en-US/web/theming)

They've been more or less on the default offering since 0.15.0 so the outbounds aren't "current"  and i'd rely on the changelogs for carrying through with them today, but I'd def check them out if for no other reason than to ensure you have full control on theming (at Toolbar, at least) and its potentials.

**Comment by longhornrumble** [comment]
Thanks.

I ended up deleting the entire project and reinstalling with Cloud Formation. I put my info in the template and everything worked!

There are a few weird behaviors:

1. When I launched the iFrame the very first time after the fresh install, the widget colors took on what i'd specified in the template.  I was optimistic!
2. Then, I inserted my custom CSS file. Some things took on the characteristics of the CSS file, but several things didn't. Namely the toolbar. It took on the settings of the 3.4.6_dist_vuetify.min.css file. 

3. I deleted my CSS file, hoping it would revert to the color scheme from my launch template, but no dice.

Nothing I seem to do is overriding that Vuetify file.  Any guesses? My custom CSS is the last thing loading on Index. I have !important marked too. Weird.

For your information, I have all of the various icons in the toolbar enabled. I don't need them, but you suggested turning them on, which might help troubleshoot.


![Screenshot 2024-07-04 at 3 09 31 AM](https://github.com/aws-samples/aws-lex-web-ui/assets/64515532/08959580-58fd-466b-897e-7667418756c0)

**Comment by israelias** [comment]
@longhornrumble Nice this is familiar. Can you try adding these two css blocks? It's the selectors I have for toolbar overrides in `custom-chatbot-style.css` by evolution.


```css
.v-toolbar .v-menu .v-btn__content {
    height: unset;
}

.v-toolbar__title {
    font-size: 14px !important;   
}
```

You'll need to adjust `lex-web-ui-loader-config.json` (which should be in the distribution directory) along with the css in the s3 bucket serving that /dist.

The red is pulling from ui.toolbarColor = "red". Change it to the hex string value that's the same as whichever/wherever you have it as background-color in `custom-**.css`.
Any other ui.** in that `*-config.json` with style-related toolbar stuff will override the .css.

You're gonna need to invalidate cache on the s3 every time you update/re-upload the .json and the .css.

See if that (at the very least) moves you out of red.

**Comment by longhornrumble** [comment]
Thanks! I'm happy to say that I figured it out.

1. The toolbar issue is spelled out under the readme css file. In my cutom css file, I changed:

.header .toolbar.theme--dark {
    background-color: #a29060 !important;

to

.bg-red {
    background-color: #a29060 !important;

2.  I think you'll want to change .v-toolbar__title to .v-toolbar-title (dash versus underscore is the direct replacement)

We should probably make these changes in the custom css file.

Appreciate the help!
